include/bgra8888.h

```diff
@@ -0,0 +1,98 @@
+/*
+ *  SPDX-License-Identifier: GPL-2.0-or-later
+ *
+ *  Copyright (C) 2023-2023  The DOSBox Staging Team
+ *
+ *  This program is free software; you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation; either version 2 of the License, or
+ *  (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License along
+ *  with this program; if not, write to the Free Software Foundation, Inc.,
+ *  51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+ */
+
+#ifndef DOSBOX_RGBA8888_H
+#define DOSBOX_RGBA8888_H
+
+#include <cstdint>
+
+// A class that holds the colour channels as an array of 8-bit values in memory
+// order (alsa known as byte order), regardless of endianness. The bytes in
+// memory are always Blue, Green, Red then Alpha.
```

Comment:
"Class to represent a BGRA8888 pixel stored as an array of four consecutive bytes (so the representation is endianness-agnostic). The bytes in memory are always Blue, Green, Red, and then Alpha."